src/Generating/StructGen.cpp

```diff
@@ -183,6 +183,14 @@ void cStructGenTrees::ApplyTreeImage(
 				CASE_TREE_OVERWRITTEN_BLOCKS:
 				{
 					a_ChunkDesc.SetBlockTypeMeta(itr->m_RelX, itr->m_RelY, itr->m_RelZ, itr->m_BlockType, itr->m_BlockMeta);
+					// If grass is below our tree, turn it to dirt
+					if (
+						(cBlockInfo::IsSolid(itr->m_BlockType)) &&
```

Comment:
Could it be a good idea to pull the grass-to-dirt check out of BlockDirt and use that? I remember something about there being water checks too?